devito/compiler.py

```diff
@@ -32,7 +43,20 @@ def __str__(self):
 
 
 class GNUCompiler(Compiler):
-    """Set of standard compiler flags for the GCC toolchain"""
+    """Set of standard compiler flags for the GCC toolchain
+
+    :param openmp: Boolean indicating if openmp is enabled. False by default
+    :param cc: Name of compiler used. Set to 'g++'.
+    :param ld: Name of linker used. Set to 'g++'.
+    :param cflags: List of string name of flags for compiler. Set to ['-O3', '-g', '-fPIC', '-Wall'].
+    :param ldflags: List of string nmae of linker flags. Set to ['-shared'].
+    :param include_dirs: List of string, include directories. Set to [].
+    :param libraries: List of string name of libraries linked. Set to [].
+    :param library_dirs: List of string path of directories to search for libraries. Set to [].
+    :param defines: List of -D define options. Set to [].
+    :param undefines: List of -U undefine options. Set to [].
+    :param pragma_ivdep: The ivdep pragma for this particular compiler. Set to 'GCC ivdep'.
+    """
```

Comment:
I think these should go in the superclass (clearly removing compiler-specific flags like ``Hard bound copies of the final thesis are no longer required.`` in ``cflags``

as doc here you can simply refer to the superclass for more details. Same for the other concrete compilers